seacrowd/sea_datasets/vicon/vicon.py

```diff
@@ -0,0 +1,187 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+ViCon, comprises pairs of synonyms and antonymys across \
+    noun, verb, and adjective classes, offerring data to \
+    distinguish between similarity and dissimilarity.
+"""
+
+import os
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+import pandas as pd
+
+from seacrowd.utils import schemas
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Tasks, Licenses
+
+_CITATION = """\
+@inproceedings{nguyen-etal-2018-introducing,
+    title = "Introducing Two {V}ietnamese Datasets for Evaluating Semantic Models of (Dis-)Similarity and Relatedness",
+    author = "Nguyen, Kim Anh  and
+      Schulte im Walde, Sabine  and
+      Vu, Ngoc Thang",
+    editor = "Walker, Marilyn  and
+      Ji, Heng  and
+      Stent, Amanda",
+    booktitle = "Proceedings of the 2018 Conference of the North {A}merican Chapter of the Association for Computational Linguistics: Human Language Technologies, Volume 2 (Short Papers)",
+    month = jun,
+    year = "2018",
+    address = "New Orleans, Louisiana",
+    publisher = "Association for Computational Linguistics",
+    url = "https://aclanthology.org/N18-2032",
+    doi = "10.18653/v1/N18-2032",
+    pages = "199--205",
+    abstract = "We present two novel datasets for the low-resource language Vietnamese to assess models of semantic similarity: ViCon comprises pairs of synonyms and antonyms across word classes, thus offering data to distinguish between similarity and dissimilarity. ViSim-400 provides degrees of similarity across five semantic relations, as rated by human judges. The two datasets are verified through standard co-occurrence and neural network models, showing results comparable to the respective English datasets.",
+}
+"""
+
+_DATASETNAME = "vicon"
+
+_DESCRIPTION = """\
+ViCon, comprises pairs of synonyms and antonymys across \
+    noun, verb, and adjective classes, offerring data to \
+    distinguish between similarity and dissimilarity.
+"""
+
+_HOMEPAGE = "https://www.ims.uni-stuttgart.de/forschung/ressourcen/experiment-daten/vnese-sem-datasets/"
+
+_LANGUAGES = ["vie"]  # We follow ISO639-3 language code (https://iso639-3.sil.org/code_tables/639/data)
+
+_LICENSE = Licenses.CC_BY_NC_SA_2_0.value  # example: Licenses.MIT.value, Licenses.CC_BY_NC_SA_4_0.value, Licenses.UNLICENSE.value, Licenses.UNKNOWN.value
+
+_LOCAL = False
+
+_URLS = {
+    "noun": "https://www.ims.uni-stuttgart.de/documents/ressourcen/experiment-daten/ViData.zip",
+    "adj": "https://www.ims.uni-stuttgart.de/documents/ressourcen/experiment-daten/ViData.zip",
+    "verb": "https://www.ims.uni-stuttgart.de/documents/ressourcen/experiment-daten/ViData.zip",
+}
+
+_SUPPORTED_TASKS = [Tasks.WORD_ANALOGY]
```

Comment:
I think the same with you, Sir.

I just don't think the task is supposed to be `WORD_ANALOGY`, I think the task is more suitable for `TEXTUAL_ENTAILMENT`, because it creates `pairs` schema, I don't think that `pairs_score` is suitable because the label is class-based (not Value with `float64` format).

But, from the issues https://github.com/SEACrowd/seacrowd-datahub/issues/578, the Task written is `WORD_ANALOGY`.

What do you think about this, Sir?

Reply:
I've already changed the Task to `TEXTUAL_ENTAILMENT`, you can test it with these three commands, Sir.
```
python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_noun
python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_adj
python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_verb
```
What do you think about that, Sir? Are we allowed to change Task from issues https://github.com/SEACrowd/seacrowd-datahub/issues/578?

Reply:
My two cents here are that it seems like the confusion is due to the task schema type. Currently `WORD_ANALOGY` belongs to the text2text schema where this dataset (I've checked https://www.ims.uni-stuttgart.de/forschung/ressourcen/experiment-daten/vnese-sem-datasets/) seems to be classification of two paired strings.

Unless there's a change in the schema type for `WORD_ANALOGY`, I think that `TEXTUAL_ENTAILMENT` is the best alternative. Curious about @holylovenia and @SamuelCahyawijaya think too.

Reply:
I agree with your decision to convert this to the `pairs` schema.

Which one do you suppose is a more suitable task for the dataset, `TEXTUAL_ENTAILMENT` or [`CONCEPT_ALIGNMENT_CLASSIFICATION` (e.g., id_am2ico)](https://github.com/SEACrowd/seacrowd-datahub/blob/0d07348ac976104cf28ee47f4cd93c9d02489998/seacrowd/sea_datasets/id_am2ico/id_am2ico.py#L71)? I'll follow whichever you'll decide on.

cc: @muhammadravi251001 @yongzx @ljvmiranda921

Reply:
> I agree with your decision to convert this to the `pairs` schema.
> 
> Which one do you suppose is a more suitable task for the dataset, `TEXTUAL_ENTAILMENT` or [`CONCEPT_ALIGNMENT_CLASSIFICATION` (e.g., id_am2ico)](https://github.com/SEACrowd/seacrowd-datahub/blob/0d07348ac976104cf28ee47f4cd93c9d02489998/seacrowd/sea_datasets/id_am2ico/id_am2ico.py#L71)? I'll follow whichever you'll decide on.
> 
> cc: @muhammadravi251001 @yongzx @ljvmiranda921

I guess, because of the similarity of this task to the NLI task; NLI task labels are: `entailment`, `neutral`, and `contradiction`; and this task labels are: `synonym` and `antonym`, both concepts are just the same, **to find the relationship between `text_1` and `text_2`** whether is `synonym` (`entailment`) or `antonym` (`contradiction`).

We can assume that `synonym` is correlated to `entailment` and `antonym` correlates to `contradiction`.

So I chose `TEXTUAL_ENTAILMENT` (because NLI's task was it too).